src/lib/operators/projection.cpp

```diff
@@ -120,27 +120,39 @@ std::shared_ptr<const Table> Projection::_on_execute() {
                 *referenced_dictionary_segment, [&](const auto& typed_segment) {
                   using DictionarySegmentType = std::decay_t<decltype(typed_segment)>;
 
-                  if constexpr (std::is_same_v<DictionarySegmentType, DictionarySegment<ColumnDataType>> ||
-                                std::is_same_v<DictionarySegmentType, FixedStringDictionarySegment<ColumnDataType>>) {
-                    const auto& dictionary = typed_segment.dictionary();
-
-                    auto filtered_attribute_vector = pmr_vector<ValueID::base_type>(pos_list->size());
-
-                    auto iterable = create_iterable_from_attribute_vector(typed_segment);
+                  // Write new attribute vector containing the positions given from the input_pos_list and compress it
+                  [[maybe_unused]] auto create_filtered_and_compressed_pos_list = [](const auto& input_segment,
+                                                                                     const auto& input_pos_list) {
+                    auto filtered_attribute_vector = pmr_vector<ValueID::base_type>(input_pos_list->size());
+                    auto iterable = create_iterable_from_attribute_vector(input_segment);
                     auto chunk_offset = ChunkOffset{0};
-                    iterable.with_iterators(pos_list, [&](auto it, auto end) {
+                    iterable.with_iterators(input_pos_list, [&](auto it, auto end) {
                       while (it != end) {
                         filtered_attribute_vector[chunk_offset] = it->value();
                         ++it;
                         ++chunk_offset;
                       }
                     });
+                    return std::make_shared<FixedSizeByteAlignedVector<uint32_t>>(std::move(filtered_attribute_vector));
+                  };
+
+                  if constexpr (std::is_same_v<DictionarySegmentType, DictionarySegment<ColumnDataType>>) {
+                    const auto compressed_attribute_vector =
+                        create_filtered_and_compressed_pos_list(typed_segment, pos_list);
+                    const auto& dictionary = typed_segment.dictionary();
 
-                    auto compressed_attribute_vector =
-                        compress_vector(filtered_attribute_vector, VectorCompressionType::FixedSizeByteAligned, {});
                     output_segments[column_id] = std::make_shared<DictionarySegment<ColumnDataType>>(
                         dictionary, std::move(compressed_attribute_vector),
                         referenced_dictionary_segment->null_value_id());
+                  } else if constexpr (std::is_same_v<DictionarySegmentType,  // NOLINT
+                                                      FixedStringDictionarySegment<ColumnDataType>>) {
```

Comment:
What happened to clang-tidy here? Is it drunk?

Reply:
`src/lib/operators/projection.cpp:143:  If an else has a brace on one side, it should have it on both  [readability/braces] [5]`